pkg/k8s/client/cell.go

```diff
@@ -517,10 +549,30 @@ func NewFakeClientset() (*FakeClientset, Clientset) {
 		KubernetesFakeClientset: fake.NewSimpleClientset(),
 		enabled:                 true,
 	}
+	client.KubernetesFakeClientset.Resources = resources
+	client.SlimFakeClientset.Resources = resources
+	client.CiliumFakeClientset.Resources = resources
+	client.APIExtFakeClientset.Resources = resources
+	client.trackers = map[string]k8sTesting.ObjectTracker{
+		"slim":       client.SlimFakeClientset.Tracker(),
+		"cilium":     client.CiliumFakeClientset.Tracker(),
+		"mcs":        client.MCSAPIFakeClientset.Tracker(),
+		"kubernetes": client.KubernetesFakeClientset.Tracker(),
+		"apiexit":    client.APIExtFakeClientset.Tracker(),
+	}
+
+	fd := client.KubernetesFakeClientset.Discovery().(*fakediscovery.FakeDiscovery)
+	fd.FakedServerVersion = toVersionInfo(version)
+
 	client.clientsetGetters = clientsetGetters{&client}
 	return &client, &client
 }
 
+func toVersionInfo(rawVersion string) *versionapi.Info {
+	parts := strings.Split(rawVersion, ".")
+	return &versionapi.Info{Major: parts[0], Minor: parts[1]}
```

Comment:
Is there a chance this is being called when running Cilium in a user environment? If so we should add guard here to prevent a panic on wrong input.

Reply:
Only used by the fake client currently.